contents/tutorials/api-feature-flags.md

```diff
@@ -0,0 +1,320 @@
+---
+title: How to evaluate and update feature flags with the PostHog API
+date: 2023-02-13
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/tutorial-16.png
+topics: ['feature flags']
+---
+
+Like [capturing events](/tutorials/api-capture-events), feature flags get a special POST-only public endpoint in the PostHog API, `/decide/`.  There are also endpoints to get data from, update this data, and more. This tutorial shows you how to use these endpoints to evaluate your feature flags (both boolean and multivariate), get data about them, update them, and finally, shows how you can combine these requests together.
+
+## Setting up authentication and feature flags
+
+The `decide` endpoint requires your project API key for authentication. To get this value, go to your project settings, and copy the value under "Project Variables." This is the same value you use to initialize a library.
+
+Obviously you also need to create a flag before you can evaluate it. To create a feature flag, in your PostHog instance, go to the "Feature Flags" tab, click "New feature flag," enter a key, set release conditions to 100% of users, and click save.
+
+## Request to evaluate a basic feature flag
+
+Once your feature flag is set up, the request to evaluate the flag is straightforward. All you need is the project API key, the distinct ID of the user you want to check, and the version (`v`) as a param. 
+
+Formatting all this correctly looks like this:
+
+<MultiLanguage>
+
+```bash
+curl -v -L --header "Content-Type: application/json" -d '{
+  "api_key": "<PH_PROJECT_API_KEY>",
+  "distinct_id": "ian@posthog.com"
+}' "https://app.posthog.com/decide/?v=2"
```

Comment:
Use v=3 in all examples!

Reply:
It returns all flags, with their values, and a new param, `errorComputingFlags`, if we were unable to compute some flags because of db errors etc. etc.

Reply:
Bonus: Also mention the `evaluation_reasons` endpoint if you want to know *why* a flag was evaluated as true/false.

This is the same we use on the persons screen feature flags tab

Reply:
Thanks, I wasn't sure if v=3 was ready for light of day. Added both.